htdocs/product/list.php

```diff
@@ -371,6 +371,7 @@
 if ($search_barcode) $sql .= natural_search('p.barcode', $search_barcode);
 if (isset($search_tosell) && dol_strlen($search_tosell) > 0 && $search_tosell != -1) $sql .= " AND p.tosell = ".$db->escape($search_tosell);
 if (isset($search_tobuy) && dol_strlen($search_tobuy) > 0 && $search_tobuy != -1)   $sql .= " AND p.tobuy = ".$db->escape($search_tobuy);
+if ((!isset($search_tosell) || $search_tosell === "" || $search_tosell != 0) && (!isset($search_tobuy) || $search_tobuy === "" || $search_tobuy != 0))  $sql.= " AND (p.tosell = 1 OR p.tobuy = 1)";
```

Comment:
I suggest we change the columns tosell and tobuy to boolean to avoid the clutter of "". I cannot think of a case where it is valuable to have them not defined. If there is such a case the user can just create an extrafield and use that instead. The columns are currently smallint. WDYT?